File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var map = new Map()
    var answer = nums.length / 2
    for(const n of nums){
        map.set(n, map.get(n) + 1 || 1)
    }
    if(map.size < answer) return map.size
    
    return answer
    
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int n = nums.length / 2;
        
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int num : nums){
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        if(map.size() > n){return n;}
        else {return map.size();}
    }
}